### ISSUE TITLE
spec.driver.coreRequest 0.5 not working

### ISSUE THREAD

**Issue by weicheng113** [GoogleCloudPlatform/spark-on-k8s-operator#925]
Hi, this is the issue following the [previous question](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/issues/922). it looks setting spec.driver.coreRequest to 0.5 is not working. I inspected related yaml:

1. Core Request is correct in the following yaml.
kubectl describe scheduledsparkapplications myScheduledApp
```
Spec:
  Concurrency Policy:            Forbid
  Failed Run History Limit:      3
  Schedule:                      0 0 * * *
  Successful Run History Limit:  1
  ...
    Driver:
      Core Limit:    1200m
      Core Request:  0.5 // Correct here
```
2. Core request is not get applied correctly in pod.
kubectl describe pods/myScheduledApp-1590370829543600455-driver
```
...
                spark-role=driver
                sparkoperator.k8s.io/launched-by-spark-operator=true
                sparkoperator.k8s.io/submission-id=dbb8fd6b-d193-4419-9fc2-28d9fea557d9
                ...
    Limits:
      cpu:     1200m
      memory:  896Mi
    Requests:
      cpu:     1 // Incorrect. 0.5 expected.
      memory:  896Mi
```

I tried the same to my sparkApplication(none scheduled one). Corerequest does not seem to be applied. Thanks for your time.

**Comment by liyinan926** [comment]
Forgot to mention that `spec.driver.coreRequest` is only available since Spark 3.0, which will be released soon.

**Comment by weicheng113** [comment]
@liyinan926 Great. Thanks. close now.
